sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala

```diff
@@ -41,19 +41,21 @@ trait QueryExecutionListener {
    * A callback function that will be called when a query executed successfully.
    *
    * @param funcName name of the action that triggered this query.
+   * @param executionId identifier of a query execution.
    * @param qe the QueryExecution object that carries detail information like logical plan,
    *           physical plan, etc.
    * @param durationNs the execution time for this query in nanoseconds.
    *
    * @note This can be invoked by multiple different threads.
    */
   @DeveloperApi
-  def onSuccess(funcName: String, qe: QueryExecution, durationNs: Long): Unit
+  def onSuccess(funcName: String, executionId: Long, qe: QueryExecution, durationNs: Long): Unit
```

Comment:
we can't just add like this .. it breaks binary compatibility

Reply:
Yeah...That is also my concerns about this.